ceno_zkvm/src/instructions/riscv/mul.rs

```diff
@@ -330,20 +388,19 @@ impl<E: ExtensionField, I: RIVInstruction> Instruction<E> for MulhInstructionBas
 
                 // assign nonzero value (i32::MAX - rd)
                 let rd_f = i64_to_base(rd_s as i64);
-                let avoid_f = i64_to_base((1i64 << (BIT_WIDTH - 1)) - 1);
+                let avoid_f = i64_to_base(i32::MAX.into());
                 constrain_rd.assign_instance(instance, rd_f, avoid_f)?;
-                // let nonzero_val = ((1i64 << (BIT_WIDTH - 1)) - 1) - (rd_s as i64);
-                // constrain_rd.assign_instance(instance, i64_to_base(nonzero_val))?;
 
                 let prod = (rs1_s as i64).wrapping_mul(rs2 as i64);
-                ((prod as u64) % (1u64 << BIT_WIDTH)) as u32
+                // only take the low part of the product
+                prod as u32
```

Comment:
That's just `rs1.wrapping_mul(rs2)` again.

Reply:
addressed in [11df823](https://github.com/scroll-tech/ceno/pull/660/commits/11df82393415723d06aab78410cf4d8665f76254)